proposals/exception-handling/Exceptions-formal-overview.md

```diff
@@ -152,34 +153,34 @@ An absent tag address in a `catch` administrative instruction (i.e., `a? = ε`)
 ```
 F; throw x  ↪  F; throw a  (if F.module.tagaddrs[x]=a)
 
-caught{a val^n} B^l[rethrow l] end
-  ↪ caught{a val^n} B^l[val^n (throw a)] end
+caught{a val*} B^l[rethrow l] end
+  ↪ caught{a val*} B^l[val* (throw a)] end
 
-caught{a val^n} val^m end  ↪  val^m
+caught{a val0*} val* end  ↪  val*
 
 
-F; val^n (try bt instr* (catch x instr'*)* (catch_all instr''*)? end)
-  ↪  F; label_m{} (catch{a instr'*}*{ε instr''*}? val^n instr* end) end
+F; val^n (try bt instr1* (catch x instr2*)* (catch_all instr3*)? end)
+  ↪  F; label_m{} (catch{a instr2*}*{ε instr3*}? val^n instr1* end) end
   (if expand_F(bt) = [t1^n]→[t2^m] ∧ (F.module.tagaddrs[x]=a)*)
 
-catch{a? instr*}* val^m end ↪ val^m
+catch{a? instr*}* val* end ↪ val*
 
-S; F; catch{a1? instr*}{a'? instr'*}* T[val^n (throw a)] end
-  ↪  S; F; caught{a val^n} (val^n)? instr* end
+S; F; catch{a1? instr1*}{a0? instr0*}* T[val^n (throw a)] end
+  ↪  S; F; caught{a val^n} (val^n)? instr1* end
   (if (a1? = ε ∨ a1? = a) ∧ S.tags(a).type = [t^n]→[])
 
-catch{a1? instr*}{a'? instr'*}* T[val^n (throw a)] end
-  ↪ catch{a'? instr'*}* T[val^n (throw a)] end
+catch{a1? instr*}{a0? instr0*}* T[val^n (throw a)] end
+  ↪ catch{a0? instr0*}* T[val^n (throw a)] end
   (if a1? ≠ ε ∧ a1? ≠ a)
 
-catch T[val^n (throw a)] end ↪  val^n (throw a)
+catch T[val* (throw a)] end ↪  val* (throw a)
```

Comment:
Oh, I think this rule needs to look up the arity of `a`, in order to know how many values to keep, because `T` may contain an arbitrary prefix of additional values that makes the decomposition ambiguous otherwise.

Reply:
Fixed in the next commit.